extensions/ql-vscode/src/queries.ts

```diff
@@ -273,7 +280,13 @@ async function checkAndConfirmDatabaseUpgrade(qs: qsClient.QueryServerClient, db
 
   logger.log(descriptionMessage);
   // Ask the user to confirm the upgrade.
-  const shouldUpgrade = await helpers.showBinaryChoiceDialog(`Should the database ${db.databaseUri.fsPath} be upgraded?\n\n${descriptionMessage}`);
+
+  let messageLines = descriptionMessage.split('\n');
+  if (messageLines.length > MAX_MESSAGE_LINES) {
+    messageLines = messageLines.slice(0, MAX_MESSAGE_LINES);
```

Comment:
Nit: similarly I think this should be `MAX_MESSAGE_LINES - 3`.

Reply:
I was just thinking of it as a coarse-grained limit on the size of the information coming from the database upgrade script. Due to line wrapping this isn't going to be exact anyway.

Reply:
Makes sense, in that case could we rename / document that this limit is coarse-grained / a heuristic?

Reply:
:+1: